### PR TITLE
fix(cdk/menu): update docs to reflect current implementation and add correct role for triggers

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/cdk/menu/menu.md [hunk 1]
```diff
@@ -10,68 +10,65 @@ directives apply their associated ARIA roles to their host element.
...
+  "example": "cdk-menu-standalone-trigger",
+  "file": "cdk-menu-standalone-trigger-example.html"
   }) -->
 
 Most menu interactions consist of two parts: a trigger and a menu panel.
 
 #### Triggers
 
-You must add the `cdkMenuItem` and `cdkMenuTriggerFor` directives to triggers like so,
+`cdkMenuTriggerFor` can be added to any button to make it a trigger for the given menu, or any menu
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:40]:
```suggestion
You can add `cdkMenuTriggerFor` to any button to make it a trigger for the given menu, or any menu
```
nit: passive voice

File: src/cdk/menu/menu.md [hunk 2]
```diff
@@ -10,68 +10,65 @@ directives apply their associated ARIA roles to their host element.
...
+item to make it a trigger for a submenu. When adding this directive, be sure to pass a reference to
+the template containing the menu it should open. You can toggle the associated menu using a mouse
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:42]:
```suggestion
the template containing the menu it should open. Users can toggle the associated menu using a mouse
```
I think "you" in this context is the developer writing the code vs. user who's going to use the menu

File: src/cdk/menu/menu.md [hunk 3]
```diff
@@ -10,68 +10,65 @@ directives apply their associated ARIA roles to their host element.
...
+#### Menu content
 
-```html
-<ng-template cdkMenuPanel #panel="cdkMenuPanel">
-  <div cdkMenu [cdkMenuPanel]="panel">
-    <!-- some content -->
-  </div>
-</ng-template>
-```
+Menu content is created using the `cdkMenu` or `cdkMenuBar` components. Menus can be always present
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:57]:
```suggestion
You can create menu content by using the `cdkMenu` or `cdkMenuBar` components. Menus can be always present
```
nit: passive voice

File: src/cdk/menu/menu.md [hunk 4]
```diff
@@ -10,68 +10,65 @@ directives apply their associated ARIA roles to their host element.
...
+on the page (referred to as an inline menu), or shown on trigger activation (referred to as a popup
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:58]: I might structure this like
```markdown
There are two types of menu:
* _inline menus_ are always present on the page
* _pop-up menus_ can be toggled to hide or show by the user
```
(also fwiw my preference is for "pop-up" vs. "popup", but both are broadly acceptable)

File: src/cdk/menu/menu.md [hunk 5]
```diff
@@ -10,68 +10,65 @@ directives apply their associated ARIA roles to their host element.
...
 Note that Angular CDK provides no styles; you must add styles as part of building your custom menu.
 
 ### Menu Bars
 
 The `CdkMenuBar` directive follows the [ARIA menubar][menubar] spec and behaves similar to a desktop
-app menubar. It consists of at least one `CdkMenuItem` which triggers a submenu. A menubar can be
-layed out horizontally or vertically (defaulting to horizontal). If the layout changes, you must set
-the `orientation` attribute to match in order for the keyboard navigation to work properly and for
-menus to open up in the correct location.
+app menubar. It consists of at least one `CdkMenuItem` which triggers a submenu.
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:71]:
```suggestion
The `CdkMenuBar` directive follows the [ARIA menubar][menubar] spec and behaves similarly to a desktop
application menubar. Each bar consists of at least one `CdkMenuItem` that triggers a submenu.
```

File: src/cdk/menu/menu.md [hunk 6]
```diff
@@ -117,45 +113,47 @@ your keyboard.
 
 ### Menu Items
 
-Both menu and menubar elements should exclusively contain menuitem elements. This directive allows
-the items to be navigated to via keyboard interaction.
+The `cdkMenuItem` directive allows the items to be navigated to via keyboard interaction.
+A menuitem can provide some user defined action by hooking into the `cdkMenuItemTriggered` output.
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:117]:
```suggestion
The `cdkMenuItem` directive allows users to navigate menuitems via keyboard.
You can add a custom action to a menuitem with the `cdkMenuItemTriggered` output.
```

File: src/cdk/menu/menu.md [hunk 7]
```diff
@@ -117,45 +113,47 @@ your keyboard.
 
 ### Menu Items
 
-Both menu and menubar elements should exclusively contain menuitem elements. This directive allows
-the items to be navigated to via keyboard interaction.
+The `cdkMenuItem` directive allows the items to be navigated to via keyboard interaction.
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:116]: general comment: we should be consistent about "menu item" vs. "menuitem" throughout. I'd probably use "menu item" unless we're explicitly talking about the ARIA role

File: src/cdk/menu/menu.md [hunk 8]
```diff
@@ -117,45 +113,47 @@ your keyboard.
...
-CdkMenuItemRadio
+<!-- example({"example":"cdk-menu-menubar",
+              "file":"cdk-menu-menubar-example.html", 
+              "region":"file-trigger"}) -->
 
 #### Menu Item Checkboxes
 
 A `cdkMenuItemCheckbox` is a special type of menuitem that behaves as a checkbox. You can use this
-type of menuitem to toggle items on and off. An element with the `cdkMenuItemCheckbox` directive
+type of menuitem to toggle items on and off. An element with  `cdkMenuItemCheckbox` directive
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:132]:
```suggestion
type of menuitem to toggle items on and off. An element with the `cdkMenuItemCheckbox` directive
```

**Review by jelbourn — APPROVED** [review]
LGTM, mostly minor grammar nits

File: src/cdk/menu/menu.md (hunk 4 above)
Comment (mmalerba) [reply src/cdk/menu/menu.md:58]: Done, wound up having to restructure a few sections to make things flow after addressing this. May be worth another review pass over this part

File: src/cdk/menu/menu.md [hunk 9]
```diff
@@ -10,68 +10,93 @@ directives apply their associated ARIA roles to their host element.
...
+
+All type of menus should exclusively contain elements with role `menuitem`, `menuitemcheckbox`,
+`menuitemradio`, or `group`. Supporting directives that automatically apply these roles are
+discussed below.
 
 Note that Angular CDK provides no styles; you must add styles as part of building your custom menu.
 
+### Inline Menus
+
+An _inline menu_ is a menu that lives directly on the page rather than a pop-up associated with a
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:72]:
```suggestion
An _inline menu_ is a menu that lives directly on the page rather than in a pop-up associated with a
```

**Review by jelbourn — APPROVED** [review]
lgtm

One thing that occurred to me for a potential follow-up PR is documenting the CSS classes applied by the cdk menu directives

File: src/cdk/menu/menu.md [hunk 10]
```diff
@@ -10,68 +10,127 @@ directives apply their associated ARIA roles to their host element.
...
+
+#### `cdkMenuItem`
+
+`cdk-menu-item`
+: Always applied
+
+#### `cdkMenuTriggerFor`
+
+`cdk-menu-trigger`
+: Always applied
```
Comment (jelbourn) [inline src/cdk/menu/menu.md:54]: I'd probably make this a table like

```markdown
| Directive | CSS Class   | Applied... |
|-----------|-------------|------------|
| `CdkMenu` | `.cdk-menu` | Always     |
|           |             |            |
|           |             |            |
```

**Review by jelbourn — COMMENTED** [review]
Repeating here from our conversation earlier for posterity:
I think the CSS classes are useful for both inheriting the directives and for when we can do directives on host elements

**Review by amysorto — APPROVED** [review]

**Pull request merged**